### ISSUE TITLE
@Value.Default not working for with deserialization for list fields since 2.2

### ISSUE THREAD

**Issue by a-k-g** [immutables/immutables#368]
```
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;

import java.io.IOException;
import java.util.Collections;
import java.util.List;

import org.immutables.value.Value;
import org.junit.Test;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.datatype.guava.GuavaModule;

public class ValueDefaultBug {

    @Value.Immutable
    @JsonDeserialize(as = ImmutableZoo.class)
    public interface Zoo {

        @Value.Default
        default List<String> animals() {
            return Collections.singletonList("Lion");
        }
    }

    @Test
    public void check_list_is_correct_when_building_zoo() {
        Zoo testZoo = ImmutableZoo.builder().build();
        assertThat(testZoo.animals(), is(Collections.singletonList("Lion")));
    }

    @Test
    public void check_zoo_deserializes_with_correct_default_list() throws IOException {
        Zoo fromJson = new ObjectMapper().registerModule(new GuavaModule())
                                         .readValue("{}", Zoo.class);
        assertThat(fromJson.animals(), is(Collections.singletonList("Lion")));
    }
}
```

Since 2.2, with the changes in @Value.Default for collection, it seems that default values are not getting honored during deserialization. The second test fails.

The culprit seems to be the following generated code:
```
@Deprecated
@JsonDeserialize
static final class Json implements ValueDefaultBug.Zoo {
    List<String> animals = ImmutableList.of();
```

Prior to 2.2, this would be:
```
@Deprecated
@JsonDeserialize
static final class Json implements ValueDefaultBug.Zoo {
    @Nullable List<String> animals;
```

Since the property is initialized to an empty list, the default value isn't getting used.

**Comment by elucash** [comment]
This is definitely an omission. Fixed in 2.2.8-SNAPSHOT. Will be released soon

**Comment by a-k-g** [comment]
Thanks for the swift fix!
